### PR TITLE
Fe feat/helper function null

### LINKED ISSUE #52: Helper functions to handle null values

### PR DESCRIPTION
**Pull request by andreasorensen**
# Description

- created `handleNoLookAlikes` helper function to conditionally render either the lookAlikes or "not provided"

- created `handleNullCharacteristics` helper function to deal with null values in characteristics to return 'not provided'

- updated DetailsPage to render changes

Closes #52 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] Testing
- [ ] Refactoring

# How Has This Been Tested?

It has not yet.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
 
+const handleNoLookAlikes = lookAlikes => {
+  return lookAlikes && lookAlikes.length
+  ? lookAlikes.map(lookAlike => lookAlike.name).join(', ') 
+  : "not provided";
+}
+
+const handleNullCharacteristics = (characteristic) => {
+  return characteristic || "not provided";
+};
```
Comment (Elise-Jones) [inline src/utils.js:15]: I love how dynamic these are great handler functions.

**Review by Elise-Jones — APPROVED** [review]

File: src/utils.js [hunk 2]
```diff
@@ -1,4 +1,13 @@
-export const formatProbability = (decimal) => {
+const formatProbability = (decimal) => {
   return Math.round(decimal * 100);
 };
 
+const handleNoLookAlikes = mushroom.lookAlikes && mushroom.lookAlikes.length
```
Comment (jmartin777) [inline src/utils.js]: boom

File: src/Components/DetailsPage/DetailsPage.js [hunk 3]
```diff
@@ -42,24 +47,22 @@ const DetailsPage = () => {
               <h3>Characterisics:</h3>
               <p> Hymenium Type: {characteristics.hymeniumType}</p>
               <p> Stipe Character: {characteristics.stipeCharacter}</p>
-              <p> Spore Print Color {characteristics.sporePrintColor}</p>
```
Comment (jmartin777) [inline src/Components/DetailsPage/DetailsPage.js:44]: great job catching these!

File: src/Components/DetailsPage/DetailsPage.js [hunk 4]
```diff
@@ -45,12 +39,12 @@ const DetailsPage = () => {
             </div>
             <div className="taxonomy">
               <h3>Characterisics:</h3>
-              <p> Hymenium Type: {characteristics.hymeniumType}</p>
-              <p> Stipe Character: {characteristics.stipeCharacter}</p>
-              <p> Spore Print Color: {characteristics.sporePrintColor}</p>
-              <p> Mushroom Cap Shape: {characteristics.mushroomCapShape}</p>
-              <p> Hymenium Attachment: {characteristics.hymeniumAttachment}</p>
-              <p> Mushroom Ecological Type: {characteristics.mushroomEcologicalType}</p>
+              <p> Hymenium Type: {handleNullCharacteristics(characteristics.hymeniumType)}</p>
```
Comment (jmartin777) [inline src/Components/DetailsPage/DetailsPage.js:42]: thank you!

**Review by jmartin777 — APPROVED** [review]
Looking good!!

**Pull request merged**